.github/workflows/mypy.yml

```diff
@@ -72,3 +72,21 @@ jobs:
     - name: Run Mypy
       run: |
         spin mypy
+  # This job checks for suspicious C API usage that may break nogil compatibility.
+  # It scans PR diffs for dangerous borrowed-referenced calls and fails if any are found.
+  lint-capi:
+    # To enable this workflow on a fork, comment out:
+    if: github.repository == 'numpy/numpy'
+    name: "C API Borrowed Ref Lint"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
```

Comment:
This needs a pinned hash (if this code is kept); please copy it exactly from the `actions/checkout` invocation ~30 lines up.